### PR TITLE
implemant registration view

### PR DESCRIPTION
**Pull request by ValentynLiashenko**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVPYTTTN-9)

## Code reviewers

- [ ] @antooa
- [ ] @HalynaDom
- [ ] @OlexaNdrus
- [ ] @pinkfoxxx
- [ ] @Rocckk
- [ ] @Rostyk1404
- [x] @Sent1nelDNoir
- [x] @Svviat

### Second Level Review

- [ ] @Blackleaves
- [x] @yevgenbehen
- [x] @lhalam

## Summary of issue

Implement user registration

### CODE REVIEW

File: iBudget/authentication/views.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def registration(request):
+    """Handling request for create new UserProfile
+
+        Args:
+            request (HttpRequest): client data
+        Returns:
+            HttpResponse object.
+    """
+    if request.method == 'POST':
```
Comment (yevgenbehen) [inline iBudget/authentication/views.py]: Maybe it is better to use `@require_http_methods(["POST"])`?

File: iBudget/utils/validators.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+"""
+This module provides function for validations.
+"""
+
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+
+MAX_LEN_PASSWORD = 16
+MIN_LEN_PASSWORD = 3
```
[the remark this answers is not in the document]
Comment (ValentynLiashenko) [reply iBudget/utils/validators.py]: good, I will change it

File: iBudget/utils/validators.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        Args:
+            password (str): The first parameter.
+
+        Returns:
+            bool: The return value. True is password valid, else False.
+
+        """
+    if not isinstance(password, str):
+        return False
+    if not MIN_LEN_PASSWORD <= len(password) <= MAX_LEN_PASSWORD:
```
[the remark this answers is not in the document]
Comment (ValentynLiashenko) [reply iBudget/utils/validators.py]: Ok. I will think about more appropriate validator

File: iBudget/authentication/models.py [hunk 4]
```diff
@@ -22,3 +22,22 @@ class UserProfile(AbstractBaseUser):
 
     objects = BaseUserManager()
     USERNAME_FIELD = 'email'
+
+    @staticmethod
+    def get_by_email(email):
+        """Static methods are similar to regular functions.
```
[the remark this answers is not in the document]
Comment (ValentynLiashenko) [reply iBudget/authentication/models.py]: Edited

File: iBudget/authentication/views.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    data = json.loads(request.body)
+    if not is_valid_registration_data(data):
+        return HttpResponse(status=400)
+    if UserProfile.get_by_email(data.get("email")):
+        return HttpResponse(status=409)
+    user = UserProfile()
+    user.email = data.get("email")
+    user.set_password(data.get("password"))
+    user.save()
```
Comment (yevgenbehen) [inline iBudget/authentication/views.py]: It looks like you have ignored my comment about safer usage if `save` method.

Comment (ValentynLiashenko) [reply iBudget/authentication/views.py]: It's done in Niclaus's task

**Pull request merged**